src/Tests/Tests/XPack/CrossClusterReplication/CrossClusterReplicationFollowTests.cs

```diff
@@ -0,0 +1,312 @@
+using System;
+using System.Linq;
+using System.Threading;
+using System.Threading.Tasks;
+using Elastic.Xunit.XunitPlumbing;
+using FluentAssertions;
+using Nest;
+using Tests.Core.ManagedElasticsearch.Clusters;
+using Tests.Core.ManagedElasticsearch.NodeSeeders;
+using Tests.Domain;
+using Tests.Framework;
+using Tests.Framework.EndpointTests.TestState;
+using Tests.Framework.Integration;
+
+namespace Tests.XPack.CrossClusterReplication
+{
+	[SkipVersion("<6.5.0", "")]
+	public class CrossClusterReplicationFollowTests : CoordinatedIntegrationTestBase<WritableCluster>
```

Comment:
There's a lot going on in this integration test! I think it may be useful to add a `<summary>` that provides a brief overview of what is being tested.

Reply:
As I understand it, each step is dependent on the "setup" of the previous one, and is asserted in its own integration test method in the class. 

If my understanding is correct, maybe we could do something similar here as to what exists for ML job API integration tests - a base class with setup steps that can be called in `IntegrationSetup` method for each deriving integration test, and a derived test class for each separate API?